pyproject.toml

```diff
@@ -0,0 +1,56 @@
+[build-system]
+requires = ["setuptools>=61.0", "pyasyncore", "cryptography", "pyoslog"] # core requirements are needed for version detection, which requires importing the script
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "emailproxy"
+authors = [
+    { name = "Simon Robinson", email = "simon@robinson.ac" }
+]
+description = "Transparently add OAuth 2.0 support to IMAP/POP/SMTP clients that don't support this authentication method."
+readme = { file = "README.md", content-type = "text/markdown" }
+license = { file = "LICENSE" }
+requires-python = ">=3.6"
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Topic :: Communications :: Email",
+    "Topic :: Communications :: Email :: Mail Transport Agents",
+    "Topic :: Communications :: Email :: Post-Office",
+    "Topic :: Communications :: Email :: Post-Office :: IMAP",
+    "Topic :: Communications :: Email :: Post-Office :: POP3",
+    "License :: OSI Approved :: Apache Software License",
+]
+dependencies = [
+    "cryptography", # for encrypting cached account tokens
+    "pyasyncore; python_version >= '3.12'", # provide the previously standard library module `asyncore`, removed in Python 3.12 (https://peps.python.org/pep-0594/)
+    "pyoslog>=0.3.0; sys_platform == 'darwin'", # macOS only: output to unified logging
+    "prompt_toolkit" # required only if using the --external-auth option in --no-gui mode
```

Comment:
I guess you don't think it's worth making an external-auth optional dependency?

Reply:
Sorry, I forgot to mention this change. Given that this isn't currently a separate script mode, I thought it was best to keep things as-is. There's also the potential for confusion because external auth in GUI mode doesn't require this dependency (but would if you decided to use the no-GUI version temporarily). So on the whole I think it's just simpler if no-GUI is the core dependencies (including `prompt_toolkit`) and GUI mode is everything in that plus anything else you might need.